From 54ae981b710324ddc190ed5eab97f265bdf402c5 Mon Sep 17 00:00:00 2001
From: Josh Boyer <jwboyer@fedoraproject.org>
Date: Thu, 3 Apr 2014 11:39:15 -0400
Subject: [PATCH] net: cpts: Add includes for ETH_HLEN and VLAN_HLEN
 definitions

If CONFIG_TI_CPTS is enabled, the ti_cpsw driver will fail to build with:

drivers/net/ethernet/ti/cpts.c: In function 'cpts_match':
drivers/net/ethernet/ti/cpts.c:266:12: error: 'ETH_HLEN' undeclared (first use in this function)
   offset = ETH_HLEN + IPV4_HLEN(data) + UDP_HLEN;
            ^
drivers/net/ethernet/ti/cpts.c:266:12: note: each undeclared identifier is reported only once for each function it appears in
drivers/net/ethernet/ti/cpts.c:276:23: error: 'VLAN_HLEN' undeclared (first use in this function)
   offset = ETH_HLEN + VLAN_HLEN;
                       ^

Add includes of if_ether.h and if_vlan.h to bring in the appropriate defines.

Signed-off-by: Josh Boyer <jwboyer@fedoraproject.org>
---
 drivers/net/ethernet/ti/cpts.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/ti/cpts.c b/drivers/net/ethernet/ti/cpts.c
index a3bbf59eaafd..2a2f82ae0c1d 100644
--- a/drivers/net/ethernet/ti/cpts.c
+++ b/drivers/net/ethernet/ti/cpts.c
@@ -19,6 +19,8 @@
  */
 #include <linux/err.h>
 #include <linux/if.h>
+#include <linux/if_ether.h>
+#include <linux/if_vlan.h>
 #include <linux/hrtimer.h>
 #include <linux/module.h>
 #include <linux/net_tstamp.h>
-- 
1.8.5.3

